src/ModuleInstantiation.cc

```diff
@@ -70,6 +70,17 @@ void IfElseModuleInstantiation::print(std::ostream &stream, const std::string &i
 	}
 }
 
+/**
+ * This is separated because PRINTB uses quite a lot of stack space
+ * and the method using it evaluate()
+ * is called often when recursive modules are evaluated.
+ * noinline is required, as we here specifically optimize for stack usage
+ * during normal operating, not runtime during error handling.
+*/
+static void __attribute__ ((noinline)) print_trace(const ModuleInstantiation *mod, const Context *ctx){
```

Comment:
We could probably use C++11 syntax here: `[[gnu::nolinline]]`, instead of the non-standard `__attribute__`.
No big deal though, as this is likely compiler-specific anyway.

Reply:
I worry about the gnu part in the alternate syntax. __attribute__ ((noinline)) is mentioned in [Attributes in Clang](https://clang.llvm.org/docs/AttributeReference.html). On the otherhand, while googling I found how chromium deals with it: [compiler_specific.h](https://chromium.googlesource.com/chromium/src/base/+/refs/heads/master/compiler_specific.h#79). On one hand, I do not like the idea of preprocessor magic, on the other hand I like the nice and clean `NOINLINE`, that can be replaced by anything the compiler likes.

I'd like to dwell on that.

Reply:
I believe the `__attribute__ ((noinline))` is GNU too, so it would not change much. I'm not sure if we should worry too much at this point as there is no MSVC build in sight and the amount of work needed for that is certainly much bigger than "just" fixing a the attribute (Yeah, I know, famous last words, but if that makes a MSVC pull request happen tomorrow, I'm happy to fix this :grinning: ).

Reply:
Yeah, to clarify: C++11 standardizes _how_ to specify attributes (the `[[ attr ]]` syntax), but `nolinline` is still a non-standard attribute.

Reply:
clang should support [[gnu::noinline]] https://clang.llvm.org/docs/AttributeReference.html#gnu-inline so let us see if our specific compiler version are able to understand it.

Reply:
> I believe the `__attribute__ ((noinline))` is GNU too, so it would not change much. I'm not sure if we should worry too much at this point as there is no MSVC build in sight and the amount of work needed for that is certainly much bigger than "just" fixing a the attribute (Yeah, I know, famous last words, but if that makes a MSVC pull request happen tomorrow, I'm happy to fix this 😀 ).

@t-paul 
I think the MSVC build is pretty much ready to merge, just not sure what the proper fix is for this as I've commented it out for now:
https://github.com/openscad/openscad/pull/2824/files#diff-f1f0fb4f6529133f175aad3e8b5917eb

Reply:
@thehans I suggest `#include "compiler_specific.h"` in order to replace  `__attribute__ ((noinline)) ` with `NOINLINE`. That way, it defaults to defining `NOINLINE` to nothing, which is at least a workaround. (and a preparation when we find a proper way to noinline for MSVC or any other compiler)

https://github.com/openscad/openscad/blob/27ce666e9c9e3f80bd8f022cf94198a816b4901a/src/compiler_specific.h#L27-L33